framework/src/actions/AddRelationshipManager.C

```diff
@@ -14,6 +14,9 @@
 registerMooseAction("MooseApp", AddRelationshipManager, "attach_geometric_rm");
 registerMooseAction("MooseApp", AddRelationshipManager, "attach_algebraic_rm");
 registerMooseAction("MooseApp", AddRelationshipManager, "attach_coupling_rm");
+registerMooseAction("MooseApp", AddRelationshipManager, "add_geometric_rm");
+registerMooseAction("MooseApp", AddRelationshipManager, "add_algebraic_rm");
+registerMooseAction("MooseApp", AddRelationshipManager, "add_coupling_rm");
```

Comment:
What are differences between "attach" and "add"?

Reply:
If you have mesh generators then RMs are "attached" to the `MeshBase`  object(s) at the same time that they are added, so no difference. When we do not have mesh generators, then attaching happens during a different phase than adding. During adding, we store the RMs in a container on `MooseApp`, and then during attaching we actually attach them to the mesh or dof-map. I *think* this was done because sometimes you cannot attach geometric relationship managers early in the simulation...like a geometric rm associated with periodic boundaries or something